rasterio/transform.py

```diff
@@ -73,14 +73,20 @@ def xy(self, row, col, z=None, offset="center", transform_method=TransformMethod
             raise AttributeError("Dataset has no {}".format(transform_method))
         return xy(transform, row, col, zs=z, offset=offset, **rpc_options)
 
-    def index(self, x, y, z=None, op=math.floor, precision=None, transform_method=TransformMethod.affine, **rpc_options):
+    def index(
+        self,
+        x,
+        y,
+        z=None,
+        op=math.floor,
+        precision=None,
+        transform_method=TransformMethod.affine,
+        **rpc_options
+    ):
         """
         Returns the (row, col) index of the pixel containing (x, y) given a
         coordinate reference system.
 
```

Comment:
Consider adding `.. deprecated:: 1.3.0 precision`

Reply:
I'm concerned about that being misunderstood as deprecation of the entire method.